### PR TITLE
Ajax Scraping Working for Firefox after the Overhaul

### PR DESCRIPTION
**Pull request by TylerMillis**

### CODE REVIEW

File: src/site_adapters/airbnb.ts [hunk 1]
```diff
@@ -5,9 +5,9 @@ import { createDomScrapingAdapter } from "./domScrapingBase"
 
 const rowContainerClass = "_fhph4u"
 const rowClass = "_8ssblpx"
-const titleClass = "_1jbo9b6h"
+const titleClass = "_1c2n35az"
```
Comment (geoffreylitt) [inline src/site_adapters/airbnb.ts:8]: optionally, would be nice to just make this as a separate change to master since it's not really part of this PR

File: src/site_adapters/domScrapingBase.ts [hunk 2]
```diff
@@ -92,6 +102,20 @@ export function createDomScrapingAdapter(config:ScrapingAdapterConfig):TableAdap
...
+  // Listen to AJAX Requests
+  browser.runtime.onMessage.addListener(request => {
+    let result = config.scrapeAjax(request);
+    if(result !== undefined && result !== null){
+      scrapedAjaxRows = result;
+      result.forEach((item) => {
+        scrapedAjaxRowDict[item.id] = item.dataValues;
+      });
+    }
+    loadTable();
```
Comment (geoffreylitt) [inline src/site_adapters/domScrapingBase.ts:117]: This is resulting in a ton of table reloads on pages which do a lot of requests, since it reloads on every request, which isn't ideal. Also, it requires the `scrapeAjax` function in the adapter to specify conditional logic in the scrape function for when to run or not, which feels a bit weird to me.

I think we should reconsider a better API surface here -- maybe the adapter specifies a function for which AJAX requests to scrape, like it specifies which URLs to activate on? Or could have a convention of returning `null` when you want to ignore a request and not reload everything?

Comment (geoffreylitt) [reply src/site_adapters/domScrapingBase.ts:117]: Actually maybe I'm overcomplicating this. Maybe it's as simple as moving this `loadTable` up a line into the if block. @TylerMillis @jessquaye thoughts?

Comment (TylerMillis) [reply src/site_adapters/domScrapingBase.ts:117]: Yeah, you are right. Moving loadTable into the if block seems to solve this issue.

File: src/site_adapters/airbnb.ts (hunk 1 above)
Comment (TylerMillis) [reply src/site_adapters/airbnb.ts]: I have reversed these changes and will make them on master. Airbnb uses different classes for Chrome vs. Firefox, so I'll have to accommodate for that and might as well do that on master.
